device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -251,6 +257,12 @@ public void open(Queue<DeviceClientConfig> deviceClientConfigs, ScheduledExecuto
         this.log.debug("AMQP connection opened successfully");
     }
 
+    private void requeueSubscribeMessages()
+    {
+        messagesToSend.addAll(subscribeMessages);
+        subscribeMessages.clear();
```

Comment:
Is there any reason why we need to save the message, instead of just creating a new one each time? I'm worried about these two queues getting out of sync. I'd feel safer if we always added a new subscribe message to the messagesToSend queue instead of relying on subscribeMessages being in the right state

Reply:
This layer should know how to build the message, after all